.github/actions/setup-job/action.yml

```diff
@@ -1,22 +1,39 @@
 name: Setup Job
 description: Common setup for all jobs
+inputs:
+    node-version:
+        description: 'The Node.js version to use'
+        required: false
+        default: '20'
 runs:
     using: 'composite'
     steps:
         - name: Enable Corepack
           shell: bash
           run: corepack enable
 
-        - name: Setup Node 20
+        - name: Setup Node LTS version
           uses: actions/setup-node@v4
           with:
-              node-version: '20'
+              node-version: ${{ inputs.node-version }}
               cache: 'yarn'
               registry-url: 'https://registry.npmjs.org'
 
+        ## --- YARN CACHE --- ##
```

Comment:
would like to know more about this job and if we can leverage it for other jobs if its pulling a cache like I think it is. could improve our run time ten-fold